File: android-core/src/main/java/com/mparticle/identity/MParticleIdentityClient.java
package com.mparticle.identity;

import com.mparticle.networking.MParticleBaseClient;

public interface MParticleIdentityClient extends MParticleBaseClient {
    IdentityHttpResponse login(IdentityApiRequest request) throws Exception;

    IdentityHttpResponse logout(IdentityApiRequest request) throws Exception;

    IdentityHttpResponse identify(IdentityApiRequest request) throws Exception;

    IdentityHttpResponse modify(IdentityApiRequest request) throws Exception;
}

File: android-core/src/main/java/com/mparticle/identity/MParticleIdentityClient.kt
package com.mparticle.identity

import com.mparticle.networking.MParticleBaseClient

interface MParticleIdentityClient : MParticleBaseClient {
    @Throws(Exception::class)
    fun login(request: IdentityApiRequest): IdentityHttpResponse

    @Throws(Exception::class)
    fun logout(request: IdentityApiRequest): IdentityHttpResponse

    @Throws(Exception::class)
    fun identify(request: IdentityApiRequest): IdentityHttpResponse

    @Throws(Exception::class)
    fun modify(request: IdentityApiRequest): IdentityHttpResponse
}
